### PR TITLE
chore(search): change hover and active state colors

### LINKED ISSUE #1975: [Audit] Search

### PR DESCRIPTION
**Pull request by aledavila**
Closes #1975 

#### Changelog

**Changed**

- hover colors on active state

**Removed**

- full search hover change

### CODE REVIEW

**Review by laurenmrice — CHANGES_REQUESTED** [review]
**Changes:**

~In Chrome: When the whole search field is in focus, and you click the close icon to show the active state, the background color is correct but there shouldn't be a focus blue border around the icon as well.~ Following DAs recommendations for accessibility.

In all browsers:
- [x] There are some weird glitchy black lines popping up when the field is in focus and you hover on the close icon, if we could remove that.

**Comment by aledavila** [comment]
@dakahn pinging for accessibility audit on the close button

**Comment by dakahn** [comment]
On mouse down the X button should become focused. The slight color change on the background of the X button isn't strong enough contrast to convey this information alone. So the blue border is actually preferable -- though it is missing in the latest build of Firefox.

Thinking of a scenario where a user has entered a search -- then moved onto something else on the page -- then returned to the search and clicks the bar to give it focus and accidentally clicks the X button erasing their previous search. On mouse down if the button was more obvious (a blue border or heightened contrast) then they could correct their click by dragging their pointer off the button before mousing up.

**Comment by laurenmrice** [comment]
okay cool, I'm fine with that for close button, thanks DA! Ale let me know when you resolve the black line glitches and we will be good to go!

File: src/components/search/_search.scss [hunk 1]
```diff
@@ -285,16 +281,21 @@
     width: rem(40px);
     right: 0;
     fill: $icon-01;
+    border: 2px solid transparent;
+    border-left: 0px;
```
Comment (joshblack) [inline src/components/search/_search.scss]:
```suggestion
    border-left: 0;
```

File: src/components/search/_search.scss [hunk 2]
```diff
@@ -285,16 +281,21 @@
...
+    &:active {
+      @include focus-outline('outline');
+    }
   }
 
   .#{$prefix}--search-input:focus ~ .#{$prefix}--search-close:hover {
-    border: 2px solid transparent;
-    background-clip: content-box;
+    border: 2px solid $focus;
+    border-left: 0px;
```
Comment (joshblack) [inline src/components/search/_search.scss]:
```suggestion
    border-left: 0;
```

**Review by joshblack — CHANGES_REQUESTED** [review]

**Review by laurenmrice — CHANGES_REQUESTED** [review]
The only thing I am seeing now is for the active state of the close icon, the focus is showing up which is good, but the background color beneath it should not be the hover-ui color token it should be the  'selected-ui' color token.

File: src/components/search/_search.scss (hunk 2 above)
Comment (aledavila) [reply src/components/search/_search.scss]: Done !

**Review by dakahn — APPROVED** [review]
Great work! 🐬🐬🐬

**Review by dakahn — APPROVED** [review]
👍🏽

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Okay one more change, the close icon for the 48px search needs to be 20px instead of 16px.

**Review by lovemecomputer — APPROVED** [review]
cool good work!! i know the cross-browser stuff on this one was weirdly tricky.

**Review by laurenmrice — APPROVED** [review]
Everything looks good to me ! 👍

**Review by joshblack — APPROVED** [review]

File: src/components/search/_search.scss [hunk 3]
```diff
@@ -268,6 +260,11 @@
     transform: translate(-50%, -50%);
   }
 
+  // .#{$prefix}--search--xl .#{$prefix}--search-close {
```
Comment (joshblack) [inline src/components/search/_search.scss]: Could we remove the commented changes?

**Review by joshblack — CHANGES_REQUESTED** [review]

**Comment by aledavila** [comment]
@joshblack done

**Review by joshblack — APPROVED** [review]

**Pull request merged**